File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/socialmedia/ReactionEntity.java
package com.github.dedis.popstellar.repository.database.socialmedia;

import androidx.annotation.NonNull;
import androidx.room.*;

import com.github.dedis.popstellar.model.Immutable;
import com.github.dedis.popstellar.model.objects.Reaction;
import com.github.dedis.popstellar.model.objects.security.MessageID;

@Entity(tableName = "reactions")
@Immutable
public class ReactionEntity {

  @PrimaryKey
  @ColumnInfo(name = "reaction_id")
  @NonNull
  private final MessageID reactionId;

  @ColumnInfo(name = "chirp_id", index = true)
  @NonNull
  private final MessageID chirpId;

  @ColumnInfo(name = "reaction")
  @NonNull
  private final Reaction reaction;

  public ReactionEntity(
      @NonNull MessageID reactionId, @NonNull MessageID chirpId, @NonNull Reaction reaction) {
    this.reactionId = reactionId;
    this.chirpId = chirpId;
    this.reaction = reaction;
  }

  @Ignore
  public ReactionEntity(@NonNull Reaction reaction) {
    this(reaction.getId(), reaction.getChirpId(), reaction);
  }

  @NonNull
  public MessageID getReactionId() {
    return reactionId;
  }

  @NonNull
  public MessageID getChirpId() {
    return chirpId;
  }

  @NonNull
  public Reaction getReaction() {
    return reaction;
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/socialmedia/ReactionEntity.kt
package com.github.dedis.popstellar.repository.database.socialmedia

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.github.dedis.popstellar.model.Immutable
import com.github.dedis.popstellar.model.objects.Reaction
import com.github.dedis.popstellar.model.objects.security.MessageID

@Entity(tableName = "reactions")
@Immutable
class ReactionEntity(
    @field:ColumnInfo(name = "reaction_id") @field:PrimaryKey val reactionId: MessageID,
    @field:ColumnInfo(name = "chirp_id", index = true) val chirpId: MessageID,
    @field:ColumnInfo(name = "reaction") val reaction: Reaction
) {

  @Ignore constructor(reaction: Reaction) : this(reaction.id, reaction.chirpId, reaction)
}
